wordpress-mobile/WordPress-Android
# Feature/14705 day selection screen part two

Fixes #14705 

This PR adds the remaining view to the day selection screen. In addition it fixes a few UI bugs and adds a few improvements.

- The bottom button is now only disabled when the screen is entered after a new post is created and the selection is empty. In this case the bottom button says "Notify me". If the user enters the day selection from the site settings, the button is always enabled and the button says "Update".
- The days are redrawn based on a diff which makes the UI smoother.
- The primary button is moved outside of the recycler view and always stays visible.
- The label in site settings is fixed

To test:
- Clear app data and log in
- Make sure the flag is turned on
- Create and publish a post
- Notice the Prologue is visible
- Click on Continue
- Notice the day selection is visible
- Notice the primary button is disabled
- Rotate your screen
- Notice the primary button is still visible
- Select a few days
- Notice the primary button is now enabled
- Notice a label appears bellow the day buttons
- Deselect the buttons 
- The label disappears
- Select the buttons again
- Confirm the selection
- Go to site settings
- Notice the "Blogging goals" item has the right amount of days selected

<img width=320 src=https://user-images.githubusercontent.com/1079756/121687051-2e0cf780-cac2-11eb-9bef-dd9fc3305043.png>


## Regression Notes
1. Potential unintended areas of impact
- None

PR submission checklist:
